From 9452e00747348535e8c7e87a9edd4d0f1a2df5b0 Mon Sep 17 00:00:00 2001
From: Cody P Schafer <dev@codyps.com>
Date: Mon, 29 Oct 2018 22:40:51 -0400
Subject: [PATCH 2/3] make: make all dirs required in install

When using DESTDIR, it's possible (and typical) that none of the
directories in the target will exist. This requires that we create all
directories needed for our files.

This adds the directories that were omitted.
---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 54c992b..7dd3536 100644
--- a/Makefile
+++ b/Makefile
@@ -281,6 +281,10 @@ install-strip: build strip_options install
 
 install: build
 	mkdir -p $(DESTDIR)$(PREFIX)${includedir}
+	mkdir -p $(DESTDIR)$(PREFIX)${bindir}
+	mkdir -p $(DESTDIR)$(PREFIX)${libdir}
+	mkdir -p $(DESTDIR)$(PREFIX)${man1dir}
+	mkdir -p $(DESTDIR)$(PREFIX)${man3dir}
 	$(INSTALL_DATA) ${INSTALL_OPTS} ${MQTTLIB_C_TARGET} $(DESTDIR)${libdir}
 	$(INSTALL_DATA) ${INSTALL_OPTS} ${MQTTLIB_CS_TARGET} $(DESTDIR)${libdir}
 	$(INSTALL_DATA) ${INSTALL_OPTS} ${MQTTLIB_A_TARGET} $(DESTDIR)${libdir}
-- 
2.26.2
